x/bridge/observer/observer.go

```diff
@@ -0,0 +1,101 @@
+package keeper
+
+import (
+	"context"
+
+	abcitypes "github.com/cometbft/cometbft/abci/types"
+	"github.com/cometbft/cometbft/libs/log"
+	"github.com/cometbft/cometbft/libs/pubsub/query"
+	rpchttp "github.com/cometbft/cometbft/rpc/client/http"
+	coretypes "github.com/cometbft/cometbft/rpc/core/types"
+	comettypes "github.com/cometbft/cometbft/types"
+)
+
+type Observer struct {
+	logger        log.Logger
+	tmRpc         *rpchttp.HTTP
+	stopChan      chan struct{}
+	eventsOutChan chan<- abcitypes.Event
+}
+
+func NewObserver(logger log.Logger, rpcUrl string, eventsOutChan chan<- abcitypes.Event) (Observer, error) {
+	rpc, err := rpchttp.New(rpcUrl, "/websocket")
+	if err != nil {
+		return Observer{}, err
+	}
+
+	return Observer{
+		logger:        logger,
```

Comment:
Cosmos logger offers custom fields, let's add them
```suggestion
		logger:        logger.With("module", "observer"),
```
And I would also suggest adding the module name (ei `observer`) to the dedicated constants list